### PR TITLE
Return proper error when user tries to enable transactions

### PR DESCRIPTION
**Pull request by findepi**
Transactions are not supported, so `setAutoCommit(false)` should fail,
as it does not provide the isolation or rollback-ability requested by
the caller.

`commit()` and `rollback()` should fail in auto-commit mode (this is
what PostgreSQL JDBC driver does).

### CODE REVIEW

**Review by zhicwu — CHANGES_REQUESTED** [review]

**Comment by zhicwu** [comment]
Thanks @findepi. We probably need an option to make it acting like a full featured driver, so that it will work with more clients.

**Comment by zhicwu** [comment]
Thanks @findepi. We probably need an option to make it acting like a full featured driver, so that it will work with more clients.

File: clickhouse-jdbc/src/main/java/ru/yandex/clickhouse/ClickHouseConnectionImpl.java [hunk 1]
```diff
@@ -209,22 +209,25 @@ public String nativeSQL(String sql) throws SQLException {
...
     @Override
     public void commit() throws SQLException {
-
+        throw new SQLException("Cannot commit when auto-commit is enabled");
     }
 
     @Override
     public void rollback() throws SQLException {
-
+        throw new SQLException("Cannot commit when auto-commit is enabled");
```
[the remark this answers is not in the document]
Comment (findepi) [reply clickhouse-jdbc/src/main/java/ru/yandex/clickhouse/ClickHouseConnectionImpl.java]: thanks, changed

**Comment by findepi** [comment]
thanks for a quick merge!

**Pull request merged**